src/libs/actions/IOU.ts

```diff
@@ -1053,28 +1053,35 @@ function getUpdateMoneyRequestParams(
         // Should only update if the transaction matches the currency of the report, else we wait for the update
         // from the server with the currency conversion
         let updatedMoneyRequestReport = {...iouReport};
-        if (updatedTransaction?.currency === iouReport?.currency && updatedTransaction?.modifiedAmount) {
-            const diff = TransactionUtils.getAmount(transaction, true) - TransactionUtils.getAmount(updatedTransaction, true);
-            if (ReportUtils.isExpenseReport(iouReport) && typeof updatedMoneyRequestReport.total === 'number') {
-                updatedMoneyRequestReport.total += diff;
-            } else {
-                updatedMoneyRequestReport = iouReport
-                    ? IOUUtils.updateIOUOwnerAndTotal(iouReport, updatedReportAction.actorAccountID ?? -1, diff, TransactionUtils.getCurrency(transaction), false)
-                    : {};
-            }
-
-            updatedMoneyRequestReport.cachedTotal = CurrencyUtils.convertToDisplayString(updatedMoneyRequestReport.total, updatedTransaction.currency);
-            optimisticData.push({
-                onyxMethod: Onyx.METHOD.MERGE,
-                key: `${ONYXKEYS.COLLECTION.REPORT}${iouReport?.reportID}`,
-                value: updatedMoneyRequestReport,
-            });
-            successData.push({
-                onyxMethod: Onyx.METHOD.MERGE,
-                key: `${ONYXKEYS.COLLECTION.REPORT}${iouReport?.reportID}`,
-                value: {pendingAction: null},
-            });
+        let diff = 0;
```

Comment:
Could you refactor this not to use `let`? Maybe we could extract a local or non-local helper function.

Reply:
I'm sorry, I wasn't clear enough. My suggestion was quite technical. I asked specifically whether we could refactor the code not to use the `let`, i.e. whether the code could be expressed without introducing a mutable variable.

Reply:
Previously, we defined `diff` like: `const diff = ...`.

I'm suggesting to keep it like this. The easiest way to keep the `const` is to introduce a helper function for calculating the difference. It can even be local and take no arguments; it's up to you.

```ts
function computeDiff() {
  if (...) {
    return ...;
  } else if (...) {
    return ...;
  } else {
    // ...
  }
}

const diff = computeDiff(...);
```

I think it's best to introduce as few mutable variables to the code as possible.

Reply:
By "local" I actually meant local in the most-inner surrounding scope. It could also be understood as module-local, as opposed to exported from a module; I didn't mean that.